transcom/mymove
# MB-16181: Add authz checks for the create mtoShipment endpoint

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16181)

## Summary

This PR adds an authz check for the POST `/mto-shipments` endpoint. It checks that the service member associated with the session matches the service member associated with the move that the shipment will be created for

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Start server and customer client.
2. Log in as a service member that has just submitted their move
3. Look at the network tab and find the value for the `MoveTaskOrderID`
4. Log out and log in as another service member to get their cookie info for postman. Load into postman based on however you set up postman.
    1. [Docs for setting up postman](https://transcom.github.io/mymove-docs/docs/tools/postman/setting-up-postman)
5. Make a post request in postman to `/mto-shipments` 
    1. Since this is going to be a post request, you'll need to [set the `X-CSRF-TOKEN` header to be the `masked_gorilla_csrf` cookie value](https://transcom.github.io/mymove-docs/docs/tools/postman/intercepting-cookies-for-postman#form-a-working-get-and-patch-request-in-postman).
    2. You will also need a valid body to send along with the request. I used the following and note the `MoveTaskOrderID` was from the non logged in user (not the user I retrieved the cookies for)
```
{
  "moveTaskOrderID": "7024c8c5-52ca-4639-bf69-dd8238308c98",
  "shipmentType": "HHG",
  "requestedPickupDate": "2006-07-19",
  "requestedDeliveryDate": "1974-06-24",
  "pickupAddress": {
    "streetAddress1": "123 Main Ave",
    "city": "Anytown",
    "state": "HI",
    "postalCode": "90210",
    "streetAddress2": "Apartment 9000",
    "streetAddress3": "Montmârtre",
    "country": "USA"
  },
  "destinationAddress": {
    "streetAddress1": "123 Main Ave",
    "city": "Anytown",
    "state": "MD",
    "postalCode": "90210",
    "streetAddress2": "Apartment 9000",
    "streetAddress3": "Montmârtre",
    "country": "USA"
  }
}
```
6. Verify you get a 403 forbidden response.
  
<img width="1422" alt="Screenshot 2023-07-13 at 11 13 34 AM" src="https://github.com/transcom/mymove/assets/67110378/ba4d78ca-638c-4079-b267-eba1b2f35153">
